### DISCUSSION TITLE
Slow server startup when using Docker

### DISCUSSION THREAD

**Discussion by chris-deluca** [mastodon/mastodon#29051] [Q&A]
I'm running Mastodon via a Docker container.  I'm currently running v4.2.5.  When I bring the container up, it sits at the line with the text: "...using keys found in /config/keys..." for a long time (today it was 16 minutes) - it has sat there for over an hour before.  Once it starts moving on in the processes, everything comes up very quickly.  There is no logged indication of what it is doing.

Any suggestions on what to look for to reduce the wait time?  

![mastodon-slow-start](https://github.com/mastodon/mastodon/assets/21092038/ae1fffeb-43ef-49ed-945a-4f02faf07f29)

**Comment by MrPistachios** [comment]
I just ran into this same issue and this post was the first result in my search, it seems its doing a check for permissions and if you have a large cache it will take a long time

I found the information here
https://github.com/linuxserver/docker-mastodon/issues/14

**Comment by chris-deluca** [comment]
I am running this command line switch in my docker-compose.yml file:  NO_CHOWN=true

The above is present in 1 of my 2 instances.  The other instance (a much smaller instance) doesn't require the setting.

P.S. I had to update my answer as 1 of my 2 instances are still using this flag.
